DjangoProject/bookmarksapp/views.py

```diff
@@ -66,3 +66,28 @@ def get_ordering(self):
             return sort
         else:
             return None
+
+
+class FolderCreateView(CreateView, FolderListView):
+    model = Folders
+    fields = ['name', ]
+    template_name = 'bookmarksapp/folderlist.html'
+
+    def dispatch(self, request, *args, **kwargs):
```

Comment:
Use `login_required` decorator https://docs.djangoproject.com/en/2.1/topics/class-based-views/intro/#decorating-the-class. Where ever you have overridden `dispatch`.